test/js-native-api/common.h

```diff
@@ -36,15 +36,14 @@
     abort();                                                                   \
   } while (0)
 
-#define NODE_API_ASSERT_BASE(env, assertion, message, ret_val)           \
-  do {                                                                   \
-    if (!(assertion)) {                                                  \
-      napi_throw_error(                                                  \
-          (env),                                                         \
-        NULL,                                                            \
-          "assertion (" #assertion ") failed: " message);                \
-      return ret_val;                                                    \
-    }                                                                    \
+#define NODE_API_ASSERT_BASE(env, assertion, message, ret_val)                 \
+  do {                                                                         \
+    napi_env napiEnv = (napi_env)env;                                          \
```

Comment:
We cannot use the `napi_throw_error` with the `node_api_basic_env`. It will cause a crash because at this point V8 has GC in an unstable state.
We must find an alternative way to report Node-API errors for the `node_api_basic_env`.

Reply:
`NODE_API_BASIC_ASSERT_RETURN_VOID` can be used instead.